src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexReadOperation.cs

```diff
@@ -646,6 +646,10 @@ private static bool WillAlwaysIncludeInResults(IndexType indexType, FieldsToFetc
                     long i = identityTracker.RegisterDuplicates(ref hasProjections, totalResults.Value, ids.AsSpan(0, read), token);
                     totalResults.Value += read; // important that this is *after* RegisterDuplicates
 
+                    // If we are going to just return count() then we don't care about anything else.
+                    if (query.IsCountQuery)
```

Comment:
That isn't sufficient for queries that want to get the total results as well, though
We have `break` below that stops us

Reply:
We want to get total results for _any_ query, don't we?. This is about filling `QueryResult.TotalResults` with correct value. We always return it so they can use it e.g. for paging.

Reply:
This is a key, if we are going to always return a valid `QueryResult.TotalResults` then we need to handle this differently. We need to decide what do we expect from Corax.

Reply:
We have to return a valid value there IMO. 

CC @ppekrol

Reply:
@redknightlois For now we need to calculate it for each query and always return valid `QueryResult.TotalResults`.

We'll do the following detection on the client side:
https://issues.hibernatingrhinos.com/issue/RavenDB-21264/Detect-that-a-query-doesnt-require-to-return-statistics-QueryResult.TotalResults-in-particular
so in the future we'll be able to skip it.